nuc_morph_analysis/analyses/linear_regression/analysis_plots.py

```diff
@@ -0,0 +1,265 @@
+    for index, row in df.iterrows():
+        if row['start_lifetime'] == 'intrinsic':
+            df.at[index, 'intrinsic_extrinsic'] = 'intrinsic'
+            df.at[index, 'start_lifetime'] = 'both'
+        elif row['start_lifetime'] == 'extrinsic':
+            df.at[index, 'intrinsic_extrinsic'] = 'extrinsic'
+            df.at[index, 'start_lifetime'] = 'both'
+
+    for target, df_target in df.groupby('target'):
+        pivot_df = df_target.pivot(index='start_lifetime', columns='intrinsic_extrinsic', values='r_squared')
+        pivot_df_std = df_target.pivot(index='start_lifetime', columns='intrinsic_extrinsic', values='stdev')
+
+        fig, ax = plt.subplots(figsize=(10, 8))
+
+        sns.heatmap(pivot_df, annot=False, cmap=cmap, ax=ax, vmin=0, vmax=0.5)
+        
+        first_element = True
+        for text_x in range(pivot_df.shape[0]):
+            for text_y in range(pivot_df.shape[1]):
+                value = pivot_df.iloc[text_x, text_y]
+                std_dev = pivot_df_std.iloc[text_x, text_y]
+                if not np.isnan(value):
+                    color = 'white' if first_element else 'black'
+                    ax.text(text_y+0.5, text_x+0.5, f'{value:.2f} ± {std_dev:.2f}', 
+                            horizontalalignment='center', 
+                            verticalalignment='center',
+                            color=color, fontsize=16)
+                    first_element = False
+        
+        ax.set_xticklabels(['','Extrinsic','Intrinsic'])
+        ax.xaxis.tick_top()
+        ax.set_yticklabels(['', 'Both', 'Lifetime', 'Start of growth'], rotation=0)
+
+        ax.set_xlabel('')
+        ax.set_ylabel('')
+        ax.tick_params(axis='both', which='both', length=0)
+        title = ax.set_title(f'Target: {get_plot_labels_for_metric(target)[1]}', loc='left')
+        title.set_position([-0.1,1])
+        save_and_show_plot(f'{figdir}{target}_prediction_r_squared_matrix_alpha_{df.alpha[0]}',  bbox_inches='tight')
+
+
+def plot_feature_contribution(coef_alpha, test_sc, perms, target, fig_height, figdir):
+    """
+    For a given target, plot feature importance for each feature in the linear model at a specified alpha.
+    Features that touch 0 are considered not important and are excluded from the plot. 
+    
+    Paramaters
+    -------
+    coef_alpha: pd.DataFrame
+        DataFrame containing the coefficient importance for each feature
+    test_sc: pd.DataFrame
+        DataFrame containing the test r2 scores
+    perms: pd.DataFrame
+        DataFrame containing the permutation test results
+    target: str
+        Prediction feature
+    fig_height: int
+        Height of the figure based on number of important features
+    save_path: str
+        Path to save the plot
+        
+    Returns
+    -------
+    Figure
+    """
+
+    alpha = coef_alpha["alpha"].unique()[0]
```

Comment:
clarify what [0] is here?

Reply:
Good question! I updated this to an argument instead that will make this more clear. 

For this plot we use the maximum alpha when the tolerance was set to 0.05. It is hardcoded in the previous step, so this line of code was just retrieving it from the dataframe. There is only one alpha value so any from the dataframe would do.